master/internal/scheduler/fitting.go

```diff
@@ -29,7 +29,8 @@ type fittingState struct {
 type FittingRequirements struct {
 	// SingleAgent specifies that the task must be located within a single agent.
 	SingleAgent bool
-	// DedicatedAgent specified that there must be no other tasks running on the agent.
+	// DedicatedAgent specifies that the task must be scheduled on dedicated agents. This is
+	// necessary in the event that host networking is used.
```

Comment:
non-blocking: Can we add `TODO (DET-3821)` to remove this and add a comment to that Jira 🙏 .